spec/models/spree/gateway/stripe_sca_spec.rb

```diff
@@ -50,6 +50,35 @@
       expect(response.success?).to eq false
       expect(response.message).to eq "Invalid payment state: succeeded"
     end
+
+    context "when payment intent state is not in 'requires_capture' state" do
+      before do
+        payment
+      end
+
+      it "succeeds if payment intent state is requires_capture" do
```

Comment:
Personally, when I spot this pattern `it "<action> if/when <condition>` to a

```rb
context `when <condition>`
  it `<action>`
  end
end
```

IMO this improves readability as I'm able to skim through `contexts` and map those into conditionals in my head. I know the code under test is expected not to behave always the same way. Also, I believe this approach also leverages RSpec's DSL.